lib/application/contracts/archethic_contract.dart

```diff
@@ -14,7 +14,12 @@ import 'package:archethic_lib_dart/archethic_lib_dart.dart' as archethic;
 import 'package:decimal/decimal.dart';
 
 class ArchethicContract with aedappfm.TransactionMixin {
-  ArchethicContract();
+  const ArchethicContract({
+    required this.apiService,
```

Comment:
Why apiService is a param here? Why don't you use the provider directly like for example `balance` provider

Reply:
`ArchethicContract` is not a provider. So it should not use `ref`.

That's why the dependencies are injected in a more traditional way.